src/matrix/login/PasswordLoginMethod.js

```diff
@@ -0,0 +1,31 @@
+/*
+Copyright 2021 The Matrix.org Foundation C.I.C.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import {LoginMethod} from "./LoginMethod.js";
+
+export class PasswordLoginMethod extends LoginMethod {
+    constructor(options) {
+        super(options);
+        this.username = options.username;
+        this.password = options.password;
+    }
+
+    async login(hsApi, deviceName) {
+        return this._platform.logger.run("passwordLogin", async log => 
```

Comment:
probably better to pass down the log item already in progress in startWithLogin, rather than to start a new log operation. The `log` argument is usually passed down as the last argument btw. Passing it down makes the /password request being logged as a child of the ongoing operation, I can show you what that looks like in the log viewer.